### PR TITLE
Fix: blocks in angular sdk and e2e setup

### PR DESCRIPTION
**Pull request by sidmohanty11**
## Description

This PR:
- adds `angular-ssr` setup for e2e
- fixes block props issue (image, video, text etc)
- allows visual editing in angular

**Loom**
https://www.loom.com/share/c6996b8f4adc4e248ffb8d68089b250d

### CODE REVIEW

File: packages/sdks/mitosis.config.cjs [hunk 1]
```diff
@@ -495,6 +495,72 @@ const ANGULAR_COMPONENT_NAMES_HAVING_HTML_TAG_NAMES = () => ({
...
+              .join(', ')}];
+            properties.forEach(prop => {
+              if (this.wrapperProps && Object.prototype.hasOwnProperty.call(this.wrapperProps, prop)) {
+                this[prop] = this.wrapperProps[prop];
+              }
+            });
+          }
+          ${code.slice(lastEndIndex)}`;
+        }
+      }
```
Comment (samijaber) [inline packages/sdks/mitosis.config.cjs:558]: can we:

- document why this is needed (we're applying properties on component initialization? why not before/elsewhere?)
- remove the duplication, something like:


```suggestion
  const propInitStr = `
              const properties = [${inputs
                .map((input) => `'${input}'`)
                .join(', ')}];
              properties.forEach(prop => {
                if (this.wrapperProps && Object.prototype.hasOwnProperty.call(this.wrapperProps, prop)) {
                  this[prop] = this.wrapperProps[prop];
                }
              });
  `
        if (code.includes('ngOnInit')) {
          code = code.replace(
            'ngOnInit() {',
            `ngOnInit() {${propInitStr}`
          );
        } else {
          const lastEndIndex = code.lastIndexOf('}');
          code = `${code.slice(0, lastEndIndex)}ngOnInit() {
            ${propInitStr}
          ${code.slice(lastEndIndex)}`;
        }
      }
```

Comment (samijaber) [reply packages/sdks/mitosis.config.cjs]: Thanks for the explanation. Can you add this as a multi-line comment for the plugin for posterity?

Comment (sidmohanty11) [reply packages/sdks/mitosis.config.cjs]: sure! added!

File: packages/sdks-tests/src/specs/index.ts [hunk 2]
```diff
@@ -143,23 +143,25 @@ export const getProps = async (args: {
   fetchOneEntry?: (opts: any) => Promise<BuilderContent | null>;
   options?: any;
   data?: 'real' | 'mock';
+  apiKey?: string;
```
Comment (samijaber) [inline packages/sdks-tests/src/specs/index.ts:146]: oh nice, good call on making this more flexible 👍🏽

**Review by samijaber — APPROVED** [review]

File: packages/sdks/src/blocks/tabs/tabs.lite.tsx [hunk 3]
```diff
@@ -35,17 +47,10 @@ export default function Tabs(props: TabsProps) {
...
               style={state.getActiveTabStyle(index)}
-              onClick={() => {
-                if (index === state.activeTab && props.collapsible) {
-                  state.activeTab = -1;
-                } else {
-                  state.activeTab = index;
-                }
-              }}
+              // eslint-disable-next-line @typescript-eslint/no-unused-vars
+              onClick={(event: any) => state.onClick(index)}
```
Comment (samijaber) [inline packages/sdks/src/blocks/tabs/tabs.lite.tsx:53]: this seems to break in svelte now

Comment (sidmohanty11) [reply packages/sdks/src/blocks/tabs/tabs.lite.tsx:53]: ahh yes, let me look into this

**Review by samijaber — APPROVED** [review]

**Pull request merged**